lesson_18/homework_18_decorators.py

```diff
@@ -0,0 +1,37 @@
+# Декоратори:
+#
+# Напишіть декоратор, який логує аргументи та результати викликаної функції.
+# Створіть декоратор, який перехоплює та обробляє винятки, які виникають в ході виконання функції.
+
+import logging
+
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+def logging_decorator(func):
+    def wrapper(*args):
+        func(*args)
+        logging.info(f"Function used as input city name: {args[0]}, city residence: {args[1]}")
+    return wrapper
+
+# # 1 way of using
+@logging_decorator
+def some_function(city_name: str, city_residents: int):
+    print(f"City name is {city_name} and has {city_residents} residents")
+# Check if decorator works
+some_function('Warsaw', 800000)
+
+# 2 way of using
+def some_function(city_name: str, city_residents: int):
+     print(f"City name is {city_name} and has {city_residents} residents")
+# Crete copy of original function
```

Comment:
please follow pep8 standard for code stylings, too many style erros
<img width="1181" alt="Screenshot 2024-10-18 at 12 53 19" src="https://github.com/user-attachments/assets/33321b85-88b7-45c4-b938-18bbcccbcba1">